File: Project 6/app/src/main/java/com/example/passdatafragmenttofragment/MainActivity.java
package com.example.passdatafragmenttofragment;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentTransaction;

import android.os.Bundle;

public class MainActivity extends AppCompatActivity implements FragmentCommunicationListener{

    private FragmentManager fragmentManager;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        fragmentManager=getSupportFragmentManager();
        launchFragmentA();
    }

    private void launchFragmentA() {
        FragmentA fragmentA=new FragmentA();
        fragmentA.setListener(this);
        FragmentTransaction fragmentTransaction=fragmentManager.beginTransaction();
        fragmentTransaction.add(R.id.flContainer, fragmentA,"fragmentA").commit();
    }

    @Override
    public void launchFragmentByPassingData(String data) {
        FragmentB fragmentB=new FragmentB();
        Bundle bundle=new Bundle();
        bundle.putString("data", data);
        fragmentB.setArguments(bundle);
        FragmentTransaction fragmentTransaction=fragmentManager.beginTransaction();
        fragmentTransaction.replace(R.id.flContainer,fragmentB,"fragmentB").addToBackStack("fragmentB").commit();

    }
}

File: Fragment_kotlin/app/src/main/java/com/example/freagment_kotlin/MainActivity.kt
package com.example.freagment_kotlin

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentTransaction
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

private lateinit var fragmentManager: FragmentManager
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        fragmentManager=supportFragmentManager
        initListener()

    }

    private fun initListener() {
        btnAnimal.setOnClickListener {
            val fragmentTransaction: FragmentTransaction=fragmentManager.beginTransaction()
           val animalFragment=AnimalFragment()
            fragmentTransaction.add(R.id.flContainer,animalFragment,"tag").addToBackStack(null).commit()
        }
    }
}